### PR TITLE
[Android] Create FileStorageManager Utility

### LINKED ISSUE #82: [Android] Create FileStorageManager Utility
**Description:**
Create a utility class for managing binary file storage (attachments such as PDFs and images) in the app's internal storage. This separates file system concerns from repository logic.

**Related Requirements:** MAT8 (Teacher), MAT15

**Tasks:**
- Create `FileStorageManager.java` utility class
- Implement `saveAttachment(materialId, attachmentId, extension, bytes)` method
  - Use predictable path pattern: `/internal/attachments/{materialId}/{attachmentId}.{ext}`
- Implement `getAttachmentFile(materialId, attachmentId)` to retrieve saved files
- Implement `deleteAttachmentsForMaterial(materialId)` for cleanup on "end lesson"
- Implement `clearAllAttachments()` for full cache clear
- Handle file I/O exceptions gracefully
- Ensure thread-safety for concurrent access
- Write unit tests with mocked file system or temporary directories

**Acceptance Criteria:**
- [x] FileStorageManager created with clear API
- [x] Save, retrieve, and delete operations functional
- [x] Predictable file path structure
- [x] Exception handling implemented
- [x] Thread-safe implementation
- [x] 100% test coverage

### PR DESCRIPTION
**Pull request by PriyaBargota**
## Summary
- Add `FileStorageManager` utility class for managing binary file storage (PDFs, images) in internal storage
- Implement thread-safe save, retrieve, and delete operations with `ReadWriteLock`
- Use predictable path pattern: `/attachments/{materialId}/{attachmentId}.{ext}`
- Add comprehensive unit tests (47 tests) covering all methods and edge cases

## Test plan
- [x] Run `./gradlew testDebugUnitTest --tests "com.manuscripta.student.utils.FileStorageManagerTest"`
- [x] Verify thread-safety tests pass under concurrent load
- [ ] Manual testing with actual PDF/image attachments in app

Closes #82

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: android/app/src/main/java/com/manuscripta/student/utils/FileStorageManager.java [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            if (!materialDir.exists() || !materialDir.isDirectory()) {
+                return null;
+            }
+
+            File[] files = materialDir.listFiles((dir, name) ->
+                    name.startsWith(attachmentId + "."));
+
+            if (files != null && files.length > 0) {
+                return files[0];
+            }
```
[the remark this answers is not in the document]
Comment (WillStephenn) [reply android/app/src/main/java/com/manuscripta/student/utils/FileStorageManager.java]: The dot acts as a delimiter that prevents the prefix collision. The pattern attachment-1. won't match attachment-10.pdf because that filename starts with attachment-10.

File: android/app/src/main/java/com/manuscripta/student/utils/FileStorageManager.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+            File attachmentFile = new File(materialDir, attachmentId + "." + extension);
+            if (!writeToFile(attachmentFile, bytes)) {
+                return null;
+            }
+            return attachmentFile;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (WillStephenn) [reply android/app/src/main/java/com/manuscripta/student/utils/FileStorageManager.java]: attachment IDs are unique, generated windows side. don't need this.
